### PR TITLE
chore: add demo site + restructure to yarn workspace

### LINKED ISSUE #37: Add demo site + move to yarn workspace
Reorganize the repo into a yarn workspace together with a demo.

### PR DESCRIPTION
**Pull request by raae**
Closes #37

### CODE REVIEW

File: demo/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "demo",
+  "private": true,
+  "description": "A simple demo to for gatsby-source-cloudinary",
```
Comment (cloudinary-raae) [inline demo/package.json]: nitpick: to - for - ?

File: demo/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "build": "gatsby build",
+    "develop": "gatsby develop",
+    "start": "gatsby develop",
+    "serve": "gatsby serve",
+    "clean": "gatsby clean",
+    "test": "echo \"Write tests! -> https://gatsby.dev/unit-testing\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/gatsbyjs/gatsby-starter-default"
```
Comment (cloudinary-raae) [inline demo/package.json]: suggestion: correct url

File: demo/src/pages/index.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { graphql } from 'gatsby';
+import { GatsbyImage, getImage } from 'gatsby-plugin-image';
+
+export default function IndexPage({ data }) {
+  return (
+    <div style={{ fontFamily: 'monospace', textAlign: 'center' }}>
+      <header>
```
Comment (cloudinary-raae) [inline demo/src/pages/index.js]: suggestion: move header inside main, remove div

File: demo/src/pages/index.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        <h1>Gatsby Source Plugin Demo</h1>
+      </header>
+
+      <main>
+        <table>
```
Comment (cloudinary-raae) [inline demo/src/pages/index.js]: praise: cool use of table

**Review by cloudinary-raae — CHANGES_REQUESTED** [review]

File: demo/src/pages/index.js (hunk 4 above)
Comment (raae) [reply demo/src/pages/index.js]: Thanks!

File: demo/package.json (hunk 2 above)
Comment (raae) [reply demo/package.json]: I'll remove it. No need for a privat demo package.

File: demo/gatsby-config.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+require('dotenv').config({
+  path: `.env.${process.env.NODE_ENV}`,
+});
+
+module.exports = {
+  siteMetadata: {
+    title: `Gatsby Source Cloudinary Starter`,
```
Comment (cloudinary-raae) [inline demo/gatsby-config.js]: nitpick: use it or remove it

**Review by cloudinary-raae — CHANGES_REQUESTED** [review]

**Review by cloudinary-raae — APPROVED** [review]

**Pull request merged**